repronim/environment/ec2environment.py

```diff
@@ -153,9 +157,93 @@ def execute_command_buffer(self):
         execution.
         """
         host = self._ec2_instance.public_ip_address
-        key_filename = self.get_config('key_filename')
+        key_filename = self['key_filename']
 
         with SSHConnector2(host, key_filename=key_filename) as ssh:
             for command in self._command_buffer:
                 self._lgr.info("Running command '%s'", command['command'])
-                self.execute_command(ssh, command['command'], command['env'])
\ No newline at end of file
+                self.execute_command(ssh, command['command'], command['env'])
+
+    def create_key_pair(self):
+        """
+        Walk the user through creating an SSH key pair that is saved to
+        the AWS platform.
+        """
+
+        def get_user_input(prompt):
```

Comment:
I think we might benefit from centralizing user interactions ... actually in the "template" copy from datalad I have already copied `repronim.ui` submodule, so you could 

```python
from repronim.ui import ui
response = ui.question(prompt)
```
benefits -- centralization of ui handling so we get uniform appearance and also handling e.g. in the case of non-interactive sessions